### ISSUE TITLE
Set genesis checkpoint roots to `Root()` in `get_forkchoice_store`

### ISSUE THREAD

**Issue by hwwhww** [ethereum/consensus-specs#2923]
> Thank @potuz for pointing it out!

In the current [`get_forkchoice_store`](https://github.com/ethereum/consensus-specs/blob/72f7b88f2a2e7d0ce0c08a36085259506c3f5df7/specs/phase0/fork-choice.md#get_forkchoice_store), it sets initial checkpoint roots to `anchor_root`:

```python
def get_forkchoice_store(anchor_state: BeaconState, anchor_block: BeaconBlock) -> Store:
    assert anchor_block.state_root == hash_tree_root(anchor_state)
    anchor_root = hash_tree_root(anchor_block)
    anchor_epoch = get_current_epoch(anchor_state)
    justified_checkpoint = Checkpoint(epoch=anchor_epoch, root=anchor_root)
    finalized_checkpoint = Checkpoint(epoch=anchor_epoch, root=anchor_root)
    ...
```

However, the genesis checkpoint roots should be `Root()`. I think we could deal with `get_current_epoch(anchor_state) == GENESIS_EPOCH` case so it would be more similar to the practice.

p.s. The non-genesis case is broken (see #2566 discussion). It would require networking calls to query the previous blocks.

**Comment by ajsutton** [comment]
Is there a particular reason the genesis checkpoint roots should be `Root()`? ie what's special about genesis vs some other state we use as the anchor?  I'm not against having a genesis specific condition here particularly but given we have to solve any issues for arbitrary states used as anchors anyway it may be redundant. Left a comment on #2566 about how Teku handles it: https://github.com/ethereum/consensus-specs/issues/2566#issuecomment-1165996834

**Comment by potuz** [comment]
> Is there a particular reason the genesis checkpoint roots should be Root()?

As far as I know this is a requirement in [the P2P spec](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/p2p-interface.md)  but it would make things much easier for us if this wasn't there and we could just reply with the `originRoot` as the checkpoint root.

**Comment by ajsutton** [comment]
You're right, from Teku:
```
            // Genesis finalized root is always ZERO because it's taken from the state and the
            // genesis block is calculated from the state so the state can't contain the actual
            // block root
            finalizedCheckpoint.getEpoch().isZero() ? Bytes32.ZERO : finalizedCheckpoint.getRoot(),
```

Definitely some odd confusion around whether the finalised checkpoint should be from the store or the head state - as seen here, they've never been guaranteed to be the same.

I suspect if we actually took the checkpoint from the state we'd be able to start from finalised state rather than needing one that matches the epoch boundary too...

**Comment by ajsutton** [comment]
Not sure if it's viable to fix the p2p spec to have the checkpoint come from the store instead of the state... It probably could be done in a two step change where clients accept either zero root or genesis block root for a while then we switch to actually sending genesis block root.

**Comment by potuz** [comment]
> Not sure if it's viable to fix the p2p spec to have the checkpoint come from the store instead of the state... It probably could be done in a two step change where clients accept either zero root or genesis block root for a while then we switch to actually sending genesis block root.

I would agree with this, this is an odd condition that surfaces from time to time in our unit tests. So that we are filled with statements like `s.ensureNotZero(root)` where the function `ensureNotZero(root)` returns `originRoot` if `root` is zero.

**Comment by leolara** [comment]
I am closing this issue because it seems stale. Please, do not hesitate to reopen it if this is a mistake
